cmd/internal/ecs/add.go

```diff
@@ -752,5 +797,79 @@ func addSecretState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowSt
 		return awf_error(errors.Wrapf(secretErr, "Failed to add an AWS secret"))
 	}
 
-	return awf_error(errors.New("Next step unimplemented!"))
+	return awf_next(modifyTaskState)
+}
+
+// Create a new revision of the task definition which includes the Akita container.
+func modifyTaskState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowState], err error) {
+	// Copy over all the state from the existing one.
+	// IDK why they didn't reuse the types.ContainerDefinition type.
+	prev := wf.ecsTaskDefinition
+	input := &ecs.RegisterTaskDefinitionInput{
+		ContainerDefinitions:    prev.ContainerDefinitions,
+		Family:                  prev.Family,
+		Cpu:                     prev.Cpu,
+		EphemeralStorage:        prev.EphemeralStorage,
+		ExecutionRoleArn:        prev.ExecutionRoleArn,
+		InferenceAccelerators:   prev.InferenceAccelerators,
+		IpcMode:                 prev.IpcMode,
+		Memory:                  prev.Memory,
+		NetworkMode:             prev.NetworkMode,
+		PidMode:                 prev.PidMode,
+		PlacementConstraints:    prev.PlacementConstraints,
+		ProxyConfiguration:      prev.ProxyConfiguration,
+		RequiresCompatibilities: prev.RequiresCompatibilities,
+		RuntimePlatform:         prev.RuntimePlatform,
+		Tags:                    wf.ecsTaskDefinitionTags,
+		TaskRoleArn:             prev.TaskRoleArn,
+		Volumes:                 prev.Volumes,
+	}
+
+	input.Tags = append(input.Tags, types.Tag{
+		Key:   aws.String(akitaCreationTagKey),
+		Value: aws.String(akitaCreationTagValue),
+	})
+
+	apiKey, apiSecret := cfg.GetAPIKeyAndSecret()
+	input.ContainerDefinitions = append(input.ContainerDefinitions, types.ContainerDefinition{
+		Name: aws.String("akita"),
+		// TODO: Cpu and Memory should be omitted for Fargate; they take their default values for EC2 if omitted.
+		// For now we can leave the defaults in place, but they might be a bit large for EC2.
+		EntryPoint: []string{"/akita", "apidump", "--project", projectFlag},
+		Environment: []types.KeyValuePair{
+			// Setting these environment variables will cause the traces to be tagged.
+			{Name: aws.String("AKITA_AWS_REGION"), Value: &wf.awsRegion},
+			{Name: aws.String("AKITA_ECS_SERVICE"), Value: &wf.ecsService},
+			{Name: aws.String("AKITA_ECS_TASK"), Value: &wf.ecsTaskDefinitionFamily},
+			{Name: aws.String("AKITA_API_KEY_ID"), Value: &apiKey},
+			{Name: aws.String("AKITA_API_KEY_SECRET"), Value: &apiSecret},
+		},
+		Essential: aws.Bool(false),
+		Image:     aws.String(akitaECRImage),
+		Secrets:   []types.Secret{},
```

Comment:
Can this be omitted? Or is the intention for there to be an explicit placeholder in case we end up using Secrets Manager? If the latter, let's say so in a comment.

Reply:
Yeah, I'll remove it.